### PR TITLE
Refactor commands

### PR DESCRIPTION
**Pull request by Booster1212**
* Refactored whole command structure, removed /src/core/server/commands folder
* Removed commands-roleplay & commands-moderator plugins
* Added core-commands plugin
* Added core-time plugin (i.e. /settime)
* Added core-weather plugin (i.e /setweather)
* Replaced playerFuncs with Athena.player.<>

### CODE REVIEW

File: src/core/plugins/core-commands/server/commands/moderator/inventory.ts [hunk 1]
```diff
@@ -0,0 +1,69 @@
...
+        Athena.player.sync.inventory(player);
+    }
+
+    @command(
+        'cleartoolbar',
+        LocaleController.get(LOCALE_KEYS.COMMAND_CLEAR_TOOLBAR, '/cleartoolbar'),
+        PERMISSIONS.ADMIN,
+    )
+
+    private static clearToolbeltCommand(player: alt.Player) {
```
Comment (LeonardSSH) [inline src/core/plugins/core-commands/server/commands/moderator/inventory.ts:29]:
```suggestion
    @command(
        'cleartoolbar',
        LocaleController.get(LOCALE_KEYS.COMMAND_CLEAR_TOOLBAR, '/cleartoolbar'),
        PERMISSIONS.ADMIN,
    )
    private static clearToolbeltCommand(player: alt.Player) {
```

File: src/core/plugins/core-commands/server/commands/moderator/noclip.ts [hunk 2]
```diff
@@ -0,0 +1,45 @@
...
+    }
+
+    static handleReset(player: alt.Player) {
+        player.spawn(player.pos.x, player.pos.y, player.pos.z, 0);
+    }
+
+    static handleCamUpdate(player, pos) {
+        player.pos = pos;
+    }
+
```
Comment (LeonardSSH) [inline src/core/plugins/core-commands/server/commands/moderator/noclip.ts:41]:
```suggestion
```

File: src/core/plugins/core-commands/server/commands/player/seatbelt.ts [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+import { command } from '../../../../../server/decorators/commands';
+import { VEHICLE_EVENTS } from '../../../../../shared/enums/vehicle';
+import { PERMISSIONS } from '../../../../../shared/flags/permissionFlags';
+import { LOCALE_KEYS } from '../../../../../shared/locale/languages/keys';
+import { LocaleController } from '../../../../../shared/locale/locale';
+
+class SeatbeltCommand {
+    @command('seatbelt',     LocaleController.get(LOCALE_KEYS.COMMAND_SEATBELT, '/seatbelt'),
+    PERMISSIONS.NONE)
+    private static handleCommand(player: alt.Player): void {
```
Comment (LeonardSSH) [inline src/core/plugins/core-commands/server/commands/player/seatbelt.ts:12]:
```suggestion
    @command('seatbelt', LocaleController.get(LOCALE_KEYS.COMMAND_SEATBELT, '/seatbelt'), PERMISSIONS.NONE)
    private static handleCommand(player: alt.Player): void {
```

File: src/core/plugins/core-time/server/index.ts [hunk 4]
```diff
@@ -0,0 +1,9 @@
+import alt from 'alt-server';
+import { PluginSystem } from '../../../server/systems/plugins';
+
+import './src/commands/timeCommands';
+
+const PLUGIN_NAME = 'Athena Time';
+PluginSystem.registerPlugin(PLUGIN_NAME, () => {
```
Comment (LeonardSSH) [inline src/core/plugins/core-time/server/index.ts:7]:
```suggestion
const PLUGIN_NAME = 'Athena Time';

PluginSystem.registerPlugin(PLUGIN_NAME, () => {
```

File: src/core/plugins/core-weather/server/index.ts [hunk 5]
```diff
@@ -0,0 +1,9 @@
+import alt from 'alt-server';
+import { PluginSystem } from '../../../server/systems/plugins';
+
+import './src/commands/weatherCommands';
+
+const PLUGIN_NAME = 'Athena Weather';
+PluginSystem.registerPlugin(PLUGIN_NAME, () => {
```
Comment (LeonardSSH) [inline src/core/plugins/core-weather/server/index.ts:7]:
```suggestion
const PLUGIN_NAME = 'Athena Weather';

PluginSystem.registerPlugin(PLUGIN_NAME, () => {
```

**Review by LeonardSSH — CHANGES_REQUESTED** [review]

**Review by LeonardSSH — APPROVED** [review]

File: src/core/shared/utility/consoleCommander.ts [hunk 6]
```diff
@@ -11,7 +11,7 @@ const Commands: ConsoleCommand = {};
  * @param {...string[]} args
  * @return {*}
  */
-function handleConsoleMessage(cmdName: string, ...args: string[]) {
```
Comment (xLuxy) [inline src/core/shared/utility/consoleCommander.ts:14]: should be
````ts
function handleConsoleMessage(cmdName: string, ...args: string[]): void

File: src/core/server/systems/chat.ts [hunk 7]
```diff
@@ -16,7 +16,7 @@ import { emitAll } from '../utility/emitHelper';
 const maxMessageLength: number = 128;
 const printCommands = false;
 let commandCount = 0;
-let commandInterval;
+let commandInterval: number;
```
Comment (xLuxy) [inline src/core/server/systems/chat.ts:19]: should be number | undefined

File: src/core/plugins/core-commands/server/index.ts [hunk 8]
```diff
@@ -0,0 +1,27 @@
...
+import './commands/moderator/vehicle';
+import './commands/moderator/wanted';
+import './commands/moderator/weapon';
+import './commands/moderator/chat';
+
+import './commands/player/chat';
+import './commands/player/death';
+import './commands/player/job';
+import './commands/player/seatbelt';
+import './commands/player/vehicle';
```
Comment (xLuxy) [inline src/core/plugins/core-commands/server/index.ts:22]: Maybe create an `index.ts` file and then just do
```ts
import './commands/player';
```
same for moderator to keep imports `clean`

**Review by xLuxy — CHANGES_REQUESTED** [review]

**Review by xLuxy — APPROVED** [review]

**Review by Stuyk — APPROVED** [review]

**Pull request merged**